ocaml/xapi/xapi_vm_clone.ml

```diff
@@ -32,6 +32,7 @@ let delete_disks rpc session_id disks =
     ) disks
 
 let wait_for_subtask ?progress_minmax ~__context task =
+  let __context = Xapi_slave_db.update_context_db __context in
```

Comment:
In this case, the context is switched over to the local DB too early. This affects DB operations (such as `Db_actions.DB_Action.Task.set_current_operations`) that are not for the event loop, which still need to use the master DB.